File: api_flask/src/api/utils/result_validation.py
class ResultValidation:
    def __init__(self):
        self.errorList = []
        self.result = {}

    def concatErrors(self, resultValidation):
        self.errorList += resultValidation.errorList

    def addError(self, tag, message, isCritical=False):
        self.errorList.append({"tag": tag, "message": message, "critical": isCritical})

    def setResult(self, result):
        self.result['data'] = result

    def hasError(self):
        return len(self.errorList) > 0

    def filterCriticalError(self, error):
        if error['critical']:
            return True
        else:
            return False

    def hasCriticalError(self):
        filteredErrors = list(filter(lambda error: error['critical'], self.errorList))
        return len(filteredErrors) > 0

    def mapErrorList(self, error):
        return {"tag": error['tag'], "message": error['message']}

    def getErrorList(self):
        return list(map(self.mapErrorList, self.errorList))

    def isResultEmpty(self):
        return self.result is None or not self.result or len(self.result) == 0

    def getResult(self):
        return self.result

    def findErrorByTags(self, tag_list):
        foundError = []
        for error in self.errorList:
            if error['tag'] in tag_list:
                foundError.append(error)
        return len(foundError) > 0

File: api/src/utils/result-validation.ts
import { 
    errorListSchemaType, 
    errorSchema, 
    resultSchemaType, 
    tagListSchemaType, 
    addResultSchemaType, 
    addResultSchema, 
    addCookieSchemaType, 
    cookieBody 
} from './resultSchema';

export class ResultValidation {


    errorList:errorListSchemaType
    result:resultSchemaType

    constructor() {
        this.errorList = [];
        this.result = {};
    }

    concatErrors(resultValidation: ResultValidation) {
        this.errorList = this.errorList.concat(resultValidation.errorList)
    }

    addError(tag: String, message: String, isCritical:Boolean = false, erro?:any) {
        const error = errorSchema.parse({tag, message, isCritical, erro})
        this.errorList.push(error);
    }

    setResult(result: addResultSchemaType) {
        const res = addResultSchema.parse(result)
        this.result['data'] = res.data;
    }

    setCookie(cookies: addCookieSchemaType){
        const cookie = cookieBody.parse(cookies)
        this.result['cookie'] ? this.result['cookie'].push(cookie) : this.result['cookie'] = [cookie]
    }

    dropCookies(){
        delete this.result['cookie']
    }

    hasError() {
        return this.errorList.length > 0;
    }

    hasCriticalError() {
        return this.errorList.filter(error => error.isCritical).length > 0;
    }

    getErrorList() {
        return this.errorList.map(error => {
            if(error.error != null){
                return {tag: error.tag, message: error.message, error: error.error} 
            }
            return {tag: error.tag, message: error.message}
        });
    }

    isResultEmpty() { //true if empty, false if not empty
        if(this.result === undefined){
            return true
        }else if(!this.result){
            return true
        }else if(this.result.data){
            return false
        }else{
            return true
        }
    }

    getResult() {
        return this.result;
    }

    clearResult(){
        this.result = {}
    }

    findErrorByTags(tagList: tagListSchemaType) {
        return this.errorList.filter(error => tagList.includes(error.tag)).length > 0;
    }
}

